adapters/adnuntius/adnuntius.go

```diff
@@ -0,0 +1,300 @@
+		if adnuntiusExt.Network != "" {
+			network = adnuntiusExt.Network
+		} else {
+			network = "default"
+		}
+
+		networkAdunitMap[network] = append(
+			networkAdunitMap[network],
+			adnAdunit{
+				AuId:     adnuntiusExt.Auid,
+				TargetId: fmt.Sprintf("%s-%s", adnuntiusExt.Auid, imp.ID),
+			})
+	}
+
+	var site string
+	if ortbRequest.Site != nil && ortbRequest.Site.Page != "" {
+		site = ortbRequest.Site.Page
+	} else {
+		site = "unknown"
+	}
+
+	for _, networkAdunits := range networkAdunitMap {
+
+		adnuntiusRequest := adnRequest{
+			AdUnits: networkAdunits,
+			Context: site,
+		}
+
+		ortbUser := ortbRequest.User
+		if ortbUser != nil {
+			ortbUserId := ortbRequest.User.ID
+			if ortbUserId != "" {
+				adnuntiusRequest.MetaData.Usi = ortbRequest.User.ID
+			}
+		}
+
+		adnJson, err := json.Marshal(adnuntiusRequest)
+		if err != nil {
+			return nil, err
+		}
+
+		requestData = append(requestData, &adapters.RequestData{
+			Method:  http.MethodPost,
+			Uri:     endpoint,
+			Body:    adnJson,
+			Headers: headers,
+		})
+
+	}
+
+	return requestData, nil
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unknown status code: %d.", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unknown status code: %d.", response.StatusCode),
+		}}
+	}
+
+	var adnResponse AdnResponse
+	if err := json.Unmarshal(response.Body, &adnResponse); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse, bidErr := generateBidResponse(&adnResponse, request)
+	if bidErr != nil {
+		return nil, bidErr
+	}
+
+	return bidResponse, nil
+}
+
+func getGDPR(request *openrtb2.BidRequest) (string, string, error) {
+	gdpr := ""
+	var extRegs openrtb_ext.ExtRegs
+	if request.Regs != nil {
+		if err := json.Unmarshal(request.Regs.Ext, &extRegs); err != nil {
+			return "", "", fmt.Errorf("failed to parse ExtRegs in Yieldlab GDPR check: %v", err)
+		}
+		if extRegs.GDPR != nil && (*extRegs.GDPR == 0 || *extRegs.GDPR == 1) {
+			gdpr = strconv.Itoa(int(*extRegs.GDPR))
+		}
+	}
+
+	consent := ""
+	if request.User != nil && request.User.Ext != nil {
+		var extUser openrtb_ext.ExtUser
+		if err := json.Unmarshal(request.User.Ext, &extUser); err != nil {
+			return "", "", fmt.Errorf("failed to parse ExtUser in Yieldlab GDPR check: %v", err)
+		}
+		consent = extUser.Consent
+	}
+
+	return gdpr, consent, nil
+}
+
+func generateBidResponse(adnResponse *AdnResponse, request *openrtb2.BidRequest) (*adapters.BidderResponse, []error) {
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(adnResponse.AdUnits))
+	currency := bidResponse.Currency
```

Comment:
It looks like you might just want to declare currency here: `var currency string`.